.github/workflows/amplify_canaries.yaml

```
@@ -156,3 +157,35 @@ jobs:
         run: |
           flutter build ios --simulator --target=integration_test/main_test.dart
           flutter test -d test integration_test/main_test.dart
+
+  log-failure-metric:
+    # Send failure data point to Cloudwatch metric CanaryTestFailure on failure
+    # See repo amplify-ci-support/github-cloudwatch-cdk for related CDK Cloudwatch setup
+    runs-on: macos-latest
+    environment: ci
+    needs: build
+    if: ${{ failure() }}
+    steps:
+      - name: Log failure data point to metric CanaryTestFailure
+        uses: ./.github/composite_actions/log_metric
+        with:
+          metric-name: CanaryTestFailure
+          value: 1
+          role-to-assume: ${{ secrets.AWS_ROLE_TO_ASSUME }}
+          aws-region: ${{ secrets.AWS_REGION }}
+
+  log-success-metric:
+    # Send success data point to Cloudwatch metric CanaryTestFailure on success
+    # See repo amplify-ci-support/github-cloudwatch-cdk for related CDK Cloudwatch setup
+    runs-on: macos-latest
+    environment: ci
+    needs: build
+    if: ${{ success() }}
+    steps:
+      - name: Log success data point to metric CanaryTestFailure
+        uses: ./.github/composite_actions/log_metric
+        with:
+          metric-name: CanaryTestFailure
+          value: 0
```

Shouldn't this be more like:

```suggestion
          metric-name: CanaryTestSuccess
          value: 1
```

What does logging a `0` value do?

---

Oh I see. Maybe link to this: https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/publishingMetrics.html#publishingZero

---

Added link.